src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/Telemetry/TelemetryReporter.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Telemetry;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Common.Telemetry
+{
+    internal static class TelemetryReporter
+    {
+        private static readonly object _guard = new();
+        private static ImmutableArray<TelemetrySession> s_telemetrySessions = ImmutableArray<TelemetrySession>.Empty;
+
+        public static void RegisterTelemetrySesssion(TelemetrySession session)
```

Comment:
Could we get a comment here explaining a bit about what a TelemetrySession might mean? For example, does opening a new document mean we should open a session for it, or is it more like a VS session? Links to the docs might be useful.

Reply:
I think Roslyn-isms are leaking here, and might be a bit overkill here. We might be better off removing this and instead having one instance of TelemetryReporter that RazorLSP creates and others can MEF import